lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -128,7 +128,16 @@ def generate_refresh_token
     end
 
     def generate_token
-      self.token = UniqueToken.generate
+      generator = Doorkeeper.configuration.access_token_generator.constantize
+
+      if generator.method(:generate).arity == 1
+        self.token = generator.generate(resource_owner_id)
+      else
+        self.token = generator.generate
+      end
```

Comment:
Rather than this conditional, what if we pass a hash (`resource_owner_id: resource_owner_id`) always, and make it an optional parameter in the default generator, keeping backwards compatibility?